src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,57 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+    private static final int DATE_FORMAT_LENGTH = 10;
+    private static final DateTimeFormatter formatter = DateTimeFormatter
+            .ofPattern("dd.MM.yyyy");
+
+    public String getSalaryInfo(String[] names, String[] data,
+                                String dateFrom, String dateTo) {
+        LocalDate fromDate = LocalDate.parse(dateFrom, formatter);
+        LocalDate toDate = LocalDate.parse(dateTo, formatter);
+
+        int[] earnings = this.getEarnings(names, data, fromDate, toDate);
+
+        //report creation
+        StringBuilder report = new StringBuilder();
+        report.append("Report for period ").append(dateFrom).append(" - ").append(dateTo);
+        for (int i = 0; i < names.length; i++) {
+            report.append(System.lineSeparator()).append(names[i]).append(" - ")
+                    .append(earnings[i]);
+        }
+
+        return report.toString();
+    }
+
+    private boolean isWithinDateRange(LocalDate currentDate, LocalDate fromDate,
+                                      LocalDate toDate) {
+        return !currentDate.isBefore(fromDate) && !currentDate.isAfter(toDate);
+    }
+
+    private int[] getEarnings(String[] names, String[] data,
+                              LocalDate fromDate, LocalDate toDate) {
```

Comment:
There are 3 loops in your code. I suggest you delete this private method. Returning `int[]` is not a good idea.
![image](https://github.com/mate-academy/jv-salary-info/assets/20326858/f20a064d-344f-4f36-8984-312278c22842)